### PR TITLE
Fix diacritics in textfields

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/3983

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Go to new story, change the locale, a different string will appear for each one. The big difference is if you toggle Zalgo text, here you can see the full range of area in which we can display a diacritic has been increased. It's gibberish, just shows where the cut off is. If you pull this locally, you can ever the change and compare to where it used to be cut off.

Note, this is safe, we'll never extend out beyond the bounds of the textfield itself, the text is overflow hidden if it tries to. With the exception of Zalgo and maybe some Tibet strings which Spectrum doesn't support anyways, the diacritics should never extend this far in the vertical range.

### CODE REVIEW

**Review by yihuiliao — COMMENTED** [review]
checked in firefox, safari, and chrome on mac and it looks good! doesn't seem to get cut off

**Review by reidbarber — APPROVED** [review]
LGTM, we should run chromatic

**Comment by snowystinger** [comment]
> LGTM, we should run chromatic

yeah, i'm running it on main right now, then I'll run it here
I didn't add a chromatic story for this because @ktabors is adding locale specific strings to chromatic in another PR

File: packages/@adobe/spectrum-css-temp/components/textfield/index.css [hunk 1]
```diff
@@ -76,7 +76,7 @@ governing permissions and limitations under the License.
   /* Change the input font styles in all browsers */
   @extend %i18nFontFamily;
   font-size: var(--spectrum-textfield-text-size);
-  line-height: var(--spectrum-textfield-text-line-height);
+  line-height: calc(var(--spectrum-textfield-height) - 2 * var(--spectrum-textfield-border-size));
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/components/textfield/index.css:79]:
```suggestion
  line-height: var(--spectrum-textfield-text-line-height);
```

**Review by devongovett — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**